src/composables/mvt-styles/mvt-styles.composable.ts

```diff
@@ -0,0 +1,205 @@
+      // }
+      // return fetch(url_get + "?key=" + label, options).then(resultsFromDB => {
+      //     return resultsFromDB.json().then(result => {
+      //         if (result.length > 0) {
+      //             const data = JSON.parse(result[0]['value']);
+      //             console.log(`Load mvt style for ${label} from database and save it to local storage`);
+      //             this.isCustomStyle = this.isCustomStyleSetter(label, true);
+      //             window.localStorage.setItem(label, JSON.stringify(data));
+      //             config.style = data['background'];
+      //             return config;
+      //         } else {
+      //             // Default style if no existing in LS or DB
+      //             console.log(`Default mvt style for ${label}`);
+      //             this.isCustomStyle = this.isCustomStyleSetter(label, false);
+      //             return config;
+      //         }
+      //     });
+      // });
+      // } else if (hasLocalStorage() && !!window.localStorage.getItem(label)) {
+      //     if (lsData.serial) {
+      //         // If there is a mvt expert style in the local storage, force parameter in the url
+      //         this.ngeoLocation_.updateParams({
+      //             'serial': JSON.stringify(lsData.serial),
+      //             'serialLayer': JSON.stringify(label)
+      //         });
+      //     }
+      //     if (lsData.medium) {
+      //         // If there is a mvt  medium config in the local storage, force parameter in the url
+      //         this.ngeoLocation_.updateParams({
+      //             'serial': JSON.stringify(lsData.medium),
+      //             'serialLayer': JSON.stringify(label)
+      //         });
+      //     }
+      //     console.log(`Load mvt style for ${label} from local storage`);
+      //     this.isCustomStyle = this.isCustomStyleSetter(label, true);
+      //     config.style = lsData.background; // Should work offline as well
+      //     return Promise.resolve(config);
+    } else {
+      // Default style if no existing in LS or DB
+      console.log(`Default mvt style for ${label}`)
+      // this.isCustomStyle = this.isCustomStyleSetter(label, false);
+
+      // temporary timeout for testing async style queries
+      return new Promise(r => setTimeout(r, 2000, config))
+      return Promise.resolve(config)
+    }
+  }
+
+  function getRoadStyleFromSimpleStyle(
+    simpleStyle: SimpleRoadStyle | null
+  ): StyleItem[] {
+    const med_road_style = bgConfigFixture().medium_default_styles
+      .road as StyleItem[]
+    // override default road style if simple style definition is available
+    if (simpleStyle) {
+      simpleStyle.colors.forEach((element: string, i: number) => {
+        med_road_style[i].color = element
+      })
+      med_road_style
+        .filter(el => el.unlocalized_label == 'Hillshade')
+        .forEach(el => {
+          el.visible = simpleStyle.hillshade
+        })
```

Comment:
i suppose it works because an element of a dictionary (visible) is modified, so the reference is compied but not the objkect

I made a simpler and more reliable version with indexof